File: Dal/Models/dto/Project.cs
namespace BPGezinswetenschappen.DAL.Models.dto
{
    public class ProjectResource
    {
        public int ProjectId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? Stage { get; set; }
        public Boolean? Active { get; set; }
        public Boolean? Supported { get; set; }
        public Boolean? Reviewed { get; set; }
        public Boolean? Approved { get; set; }
        public string? Feedback { get; set; }
        public int? StudentId { get; set; }
        public int? CoachId { get; set; }
        public int? OrganisationId { get; set; }
        public int? ProposalId { get; set; }
        public int[]? TopicIds { get; set; }

    }
}

File: Client/src/app/shared/models/project.ts
import { User } from './user';
import { Proposal } from './proposal';
import { Topic } from './topic';
import { Organisation } from './organisation';

export interface Project {
  projectId: number;
  createdAt: Date;
  updatedAt: Date;
  title: string;
  description: string;
  stage: string;
  active: boolean;
  supported: boolean;
  reviewed: boolean;
  approved: boolean;
  feedback: string;
  studentId: number;
  coachId: number;
  organisationId: number;
  proposalId: number;
  student: User;
  coach: User;
  organisation: Organisation;
  proposal: Proposal;
  topics: Topic[];
}

export interface ProjectCreateDto {
  createdAt: Date;
  updatedAt: Date;
  title: string;
  description: string;
  studentId: number;
  organisationId: number;
  proposalId: number;
  topics: Topic[];
  stage: string;
  active: boolean;
}

export interface ProjectReadDto {
  projectId: number;
  createdAt: Date;
  updatedAt: Date;
  title: string;
  description: string;
  stage: string;
  active: boolean;
  supported: boolean;
  reviewed: boolean;
  approved: boolean;
  feedback: string;
  studentId: number;
  coachId: number;
  organisationId: number;
  proposalId: number;
}

export interface ProjectUpdateDto {
  title?: string;
  description?: string;
  stage?: string;
  active?: boolean;
  supported?: boolean;
  reviewed?: boolean;
  approved?: boolean;
  feedback?: string;
  studentId?: number;
  coachId?: number;
  organisationId?: number;
  proposalId?: number;
  topicIds?: number[];
}
